pwndbg/commands/sigreturn.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import argparse
+import collections
+
+import pwndbg.color.context as C
+import pwndbg.color.memory as M
+import pwndbg.color.message
+import pwndbg.commands
+import pwndbg.gdblib.arch
+import pwndbg.gdblib.memory
+import pwndbg.gdblib.regs
+from pwndbg.lib.regs import amd64 as amd64_regset
+
+parser = argparse.ArgumentParser(description="Display the SigreturnFrame at the specific address")
+
+parser.add_argument(
+    "address", nargs="?", default=None, type=int, help="The address to read the frame from"
+)
+
+parser.add_argument(
+    "-a",
+    "--all",
+    dest="display_all",
+    action="store_true",
+    default=False,
+    help="Show all values in the frame in addition to common registers",
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+@pwndbg.commands.OnlyWithArch(["x86-64"])
+def sigreturn(address: int = None, display_all=False):
+    address = pwndbg.gdblib.regs.sp if address is None else address
+
+    arch_name = pwndbg.gdblib.arch.name
+    if arch_name == "x86-64":
```

Comment:
Is this even needed? I believe this should already be handled by `OnlyWithArch(...)`

Reply:
I see, the printing of the error message is extraneous and will never get called.